packages/babel-jest/src/index.ts

```diff
@@ -63,32 +64,57 @@ const createTransformer = (
     return babelConfig;
   }
 
+  const rootDirRegExpCache: {[rootDir: string]: RegExp} = Object.create(null);
+
+  function getRootDirRegExp(rootDir: Config.Path): RegExp {
+    if (!rootDirRegExpCache[rootDir]) {
+      rootDirRegExpCache[rootDir] = new RegExp(
+        `^${escapeStrForRegex(rootDir)}(${escapeStrForRegex(path.sep)}|$)`,
+      );
+    }
+
+    return rootDirRegExpCache[rootDir];
+  }
+
+  function getCacheKeyForBabelOptions(
+    babelOptions: PartialConfig,
+    rootDir: Config.Path,
+  ) {
+    const rootDirRegExp = getRootDirRegExp(rootDir);
+
+    // Not pretty but not coupled to a specific signature of the
+    // babel options to relativize paths
+    return JSON.stringify(babelOptions.options, (_key, value) => {
+      if (typeof value === 'string') {
+        return value.replace(rootDirRegExp, '<rootDir>$1');
```

Comment:
```suggestion
        return value.replace(rootDirRegExp, '<rootDir>');
```

Reply:
Good catch! I thought lookahead wasn't supported in Node <9 but that was lookbehind 😅😅